Content.Client/Verbs/VerbMenuElement.cs

```diff
@@ -0,0 +1,232 @@
+
+namespace Content.Client.Verbs
+{
+
+    /// <summary>
+    ///     This pop-up appears when hovering over a verb category in the context menu.
+    /// </summary>
+    public sealed class VerbCategoryPopup : ContextMenuPopup
+    {
+        public VerbCategoryPopup(VerbSystem system, IEnumerable<Verb> verbs, VerbType type, EntityUid target, bool drawOnlyIcons)
+            : base()
+        {
+            // Do any verbs have icons? If not, don't bother leaving space for icons in the pop-up.
+            var drawVerbIcons = false;
+            foreach (var verb in verbs)
+            {
+                if (verb.Icon != null)
+                {
+                    drawVerbIcons = true;
+                    break;
+                }
+            }
+
+            // If no verbs have icons. we cannot draw only icons
+            if (drawVerbIcons == false)
+                drawOnlyIcons = false;
+
+            // If we are drawing only icons, show them side by side
+            if (drawOnlyIcons)
+                List.Orientation = LayoutOrientation.Horizontal;
+
+            foreach (var verb in verbs)
+            {
+                AddToMenu(new VerbButton(system, verb, type, target, drawVerbIcons));
+            }
+        }
+    }
+
+    public sealed class VerbButton : BaseButton
+    {
+        public VerbButton(VerbSystem system, Verb verb, VerbType type, EntityUid target, bool drawIcons = true) : base()
+        {
+            Disabled = verb.Disabled;
+            ToolTip = verb.Tooltip;
+            TooltipDelay = 0.5f;
+
+            var buttonContents = new BoxContainer { Orientation = LayoutOrientation.Horizontal };
+
+            // maybe draw verb icons
+            if (drawIcons)
+            {
+                TextureRect icon = new()
+                {
+                    MinSize = (32, 32),
+                    Stretch = TextureRect.StretchMode.KeepCentered,
+                    TextureScale = (0.5f, 0.5f)
+                };
+
+                // Even though we are drawing icons, the icon for this specific verb may be null.
+                if (verb.Icon != null)
+                {
+                    icon.Texture = verb.Icon.Frame0();
+                }
+
+                buttonContents.AddChild(icon);
+            }
+
+            // maybe add a label
+            if (verb.Text != string.Empty)
+            {
+                var label = new RichTextLabel();
+                label.SetMessage(FormattedMessage.FromMarkupPermissive(verb.Text));
+                buttonContents.AddChild(label);
+
+                // If we added a label, also add some padding
+                buttonContents.AddChild(new Control { MinSize = (8, 0) });
+            }
+
+            AddChild(buttonContents);
+
+            if (Disabled)
+                return;
+
+            // give the button functionality!
+            OnPressed += _ =>
+            {
+                system.ContextMenuPresenter.CloseAllMenus();
```

Comment:
I've added that option. Currently only used for rotate, but I guess it could eventually also be useful for some other actions like knocking on windows or maybe attacking, generally whenever you want to repeatedly do the same action.